app/controllers/users/approvals_controller.rb

```diff
@@ -22,4 +26,15 @@ def reject
     render "users/approvals/approve"
   end
 
+  def check_user
+    unless current_user?(params[:user_id])
+      flash[:notice] = "You are signed in as #{current_user.username}"
+      if params[:origin]
+        developer = Developer.find_by(company_name: params[:company_name])
```

Comment:
@t-harps That sounds more like a config problem: You should be able to send headers with any request :)